users/service.go

```diff
@@ -237,6 +295,70 @@ func (svc usersService) identify(ctx context.Context, token string) (string, err
 	return email.GetValue(), nil
 }
 
+// CreateGroup adds a list of groups to the user identified by the provided key.
+func (svc usersService) CreateGroup(ctx context.Context, token string, group Group) (Group, error) {
+	userID, err := svc.auth.Identify(ctx, &mainflux.Token{Value: token})
+	if err != nil {
+		return Group{}, errors.Wrap(ErrUnauthorizedAccess, err)
+	}
+	group.Owner.ID = userID.Value
+	uid, err := uuidProvider.New().ID()
+	if err != nil {
+		return Group{}, errors.Wrap(ErrCreateUser, err)
+	}
+	group.ID = uid
+	return svc.groups.Save(ctx, group)
+}
+
+func (svc usersService) ListGroups(ctx context.Context, token string, offset, limit uint64, name string, meta Metadata) (GroupPage, error) {
```

Comment:
Who can assign user to the group? 
Who can remove the user from the group?
Who can delete a group? 

I saw you have owner checking only in GET. What's the idea for other operations?

Reply:
access rights will be introduced with policies

Reply:
That makes sense. I was confused with the owner checking in RetrieveAll.
Maybe policies should handle this also, I mean if you have read in policies that mean that I can retrieve all groups where I am assigned. Not sure how this should work together, the thing to think about :)